resources/api.json

```diff
@@ -1527,23 +1764,28 @@
             "fields": [
                 {
                     "name": "courses",
-                    "type": "map[string]core.EnrollmentInfo"
+                    "type": "map[string]core.EnrollmentInfo",
+                    "required": false
```

Comment:
Having the "required" key here is a bit awkward for output types.
It's mostly false (because that's the default), but what does it mean when it is true
(which could happen since a common type may be used as input and output)?
A user may think that required being false may mean the field may be optional in the output (when that probably isn't true).

You may want to just have the required field on the endpoint fields and leave it off of "types" entries.